api/migrationtarget/client.go

```diff
@@ -77,8 +77,9 @@ func (c *Client) Activate(modelUUID string) error {
 // to add the charm binary to the model specified.
 func (c *Client) UploadCharm(modelUUID string, curl *charm.URL, content io.ReadSeeker) (*charm.URL, error) {
 	args := url.Values{}
-	args.Add("series", curl.Series)
 	args.Add("schema", curl.Schema)
+	args.Add("user", curl.User)
+	args.Add("series", curl.Series)
```

Comment:
Note: no test changes here because these HTTP endpoints don't have tests! I have further changes coming here so I'll close the testing gap in the next PR.